tests/compiletest.rs

```diff
@@ -226,8 +238,15 @@ fn get_target() -> String {
 
 fn main() -> Result<()> {
     ui_test::color_eyre::install()?;
+
     let target = get_target();
 
+    if let Some(first) = std::env::args().nth(1) {
+        if first == "miri-run-dep-mode" {
```

Comment:
Should we make this a flag, `--miri-run-dep-mode`? We already don't allow filters starting with `-` before `--` anyway so this avoids any potential conflicts.